### ISSUE TITLE
Submenu navigation buttons have `aria-expanded` attribute when content is already expanded.

### ISSUE THREAD

**Issue by joedolson** [WordPress/gutenberg#72088]
### Description

In the navigation block, when the option 'open on click' is enabled, there are cases where the submenu will already be expanded, but the navigation item button still reports with `aria-expanded`. Initially the button reports as `aria-expanded="false"`, so a screen reader user will believe they need to open the menu. They will also expect to be able to close the menu, but this will not work, as the button changes state but the menu does not change visibility.

See https://core.trac.wordpress.org/ticket/64075 for an issue reported against WordPress core where this is an issue. 

A button that reports as being interactive, but does not actually do anything is a clear accessibility issue; and could create issues for some users. 

This can be resolved in two different ways:

1) Ensure that the `aria-expanded` attribute is not present on a parent item button if the related submenu is already visible.
2) Enable open on click to function on submenus. (See https://github.com/WordPress/gutenberg/issues/44346, https://github.com/WordPress/gutenberg/issues/57425, https://github.com/WordPress/gutenberg/issues/38599, https://github.com/WordPress/gutenberg/issues/65214)

In my opinion, it would make a lot more sense for submenu behavior to be consistent both inside and outside the modal. It may be valuable to have a separate control to enable this for submenus inside the modal. 

However, the current functionality is simply wrong. 

### Step-by-step reproduction instructions

See https://core.trac.wordpress.org/ticket/64075 for reproduction.

### Screenshots, screen recording, code snippet

See https://core.trac.wordpress.org/ticket/64075 screenshots and a screen recording.

### Environment info

- WordPress trunk
- Gutenberg not installed, Gutenberg 21.7.0.

### Please confirm that you have searched existing issues in the repo.

- [x] Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

- [x] Yes

### Please confirm which theme type you used for testing.

- [x] Block
- [ ] Classic
- [ ] Hybrid (e.g. classic with theme.json)
- [ ] Not sure

**Comment by jhmonroe** [comment]
@joedolson thank you for posting this issue!

I'm encountering a bug that must be related to this. If you think it's not related, I can open a separate issue

The class current-menu-item is no longer being applied correctly.

<img width="2345" height="941" alt="Image" src="https://github.com/user-attachments/assets/b7817547-77f1-4af9-af21-28cc500f2e99" />

Note that I am on the NEWS page here: https://joanf1.sg-host.com/news/

On the previous classic theme version of this site I have some CSS that applies border-bottom to the current-menu-item. It is set up here as well and does not work at all when OPEN ON CLICK is activated in the nav.
And when OPEN ON CLICK is turned off, the current-menu-item gets stuck on the second nav item with a submenu. No matter what page is visited these classes get stuck here.

I've tried everything... clearing caches, deactivating cache plugins, clearing cache on host side, etc.
When I pasted all that li data-wp-context code into Claude and asked about it, it says it comes from the interactivity API (including aria-expanded which you noted) and suggested I disable it temporarily to rule that out.

I am not a JS/API expert, but these suggestions seem interesting for someone who is...

<img width="772" height="838" alt="Image" src="https://github.com/user-attachments/assets/d055e5d9-d36d-4435-81e7-b6e5b4d3bddf" />

<img width="784" height="730" alt="Image" src="https://github.com/user-attachments/assets/447a6510-c9f5-4d7a-b0b4-a415867428a7" />

**Comment by hanneslsm** [comment]
@jhmonroe I think your issue is not related to this one here. Please open separate issue. Also, this one might be interesting for you: https://github.com/WordPress/gutenberg/issues/42299

**Comment by jhmonroe** [comment]
In case anyone else runs into my issue, it was solved by giving all the navs unique names. The interactivity API doesn't like if there are two versions of the same nav in a template part. (in my use case I have a desktop and mobile version with same nav items but different presentation settings that show/hide on different screen sizes)

I also had to recreate  both navs from scratch in Site Editor > Navigation. 

I was migrating a classic theme to block theme and there were old classes added (not by me!) that came thru under Advanced that were causing conflicts. Even after removing them manually it didn't work. There must have been some code in the Code Editor view that I didn't notice that came over from the classic theme.

Some things just don't play nice in Gutenberg unless you recreate them from scratch inside Gutenberg :-)
